File: gamelogic/src/main/java/com/alta_v2/game/gamelogic/domain/player/PlayerProcessor.java
package com.alta_v2.game.gamelogic.domain.player;

import com.alta_v2.common.Destroyable;
import com.alta_v2.mediator.serde.ActionListener;

public interface PlayerProcessor extends Destroyable {

    void actionBegin(ActionListener.ActionType action);

    void actionFinish(ActionListener.ActionType action);

}

File: gamelogic/src/main/kotlin/com/alta_v2/game/gamelogic/domain/player/PlayerProcessor.kt
package com.alta_v2.game.gamelogic.domain.player

import com.alta_v2.common.Destroyable
import com.alta_v2.mediator.serde.ActionListener.ActionType

interface PlayerProcessor : Destroyable {

    fun actionBegin(action: ActionType)

    fun actionFinish(action: ActionType)

}